src/api/ready.js

```diff
@@ -0,0 +1,54 @@
+import data from '../util/data';
+import registry from '../global/registry';
+
+function ready (element) {
+  const components = registry.find(element);
+  for (let a = 0; a < components.length; a++) {
+    if (!data(element, `lifecycle/${components[a].id}`).created) {
+      return false;
+    }
+  }
+  return true;
+}
+
+export default function (elements, callback) {
+  // If a selector is provided we convert it to to a collection of elements.
+  if (typeof elements === 'string') {
+    elements = document.querySelectorAll(elements);
+  }
+
+  // Ensure the collection is traversable.
+  const collection = elements.length === undefined ? [elements] : elements;
+  const collectionLength = collection.length;
+
+  // If there's no elements we don't do anything.
+  if (!collection.length) {
```

Comment:
`callback` is not supposed to be invoked with an empty element list in this case?